src/lib/core/utils/data-element-constraints.ts

```diff
@@ -293,16 +316,13 @@ export function flattenConstraints(
       );
     })
   );
+
   if (compatibleConstraints.length === 0)
     throw new Error(
-      'flattenConstraints: Something went wrong. No compatible constraints were found for the current set of values.'
+      'filterConstraints: Something went wrong. No compatible constraints were found for the current set of values.'
     );
-  // Combine compatible constraints into a single constraint, concatenating
```

Comment:
Yep, this was a good catch by @asizemore!

Reply:
The issue has to do with more complex combinations. For example

```
[
  { x: type: ['string'], maxNumValues: 10 },
  { x: type: ['number'], maxNumValues: 20 },
]
```

would become

```
  { x: type: ['string', 'number'], maxNumValues: 10 }
```

This would preclude string variables with > 10 values. I don't recall the specific case where this came up.

Reply:
This is the first time we're seeing multiple patterns for constraints in a single viz. I'm guessing some of this was drafted but never actually tested before now.